core/src/main/java/org/mule/endpoint/URIBuilder.java

```diff
@@ -255,25 +286,25 @@ private void appendMeta(StringBuilder buffer)
         }
     }
 
-    private OrderedQueryParameters appendAddress(StringBuilder buffer)
+    private OrderedQueryParameters appendAddress(StringBuilder buffer, Transformer tokenProcessor, Transformer tokenEncoder)
```

Comment:
The complexity of the transformations, having the expression evaluator in the middle, would make such thing viable only by duplicating the methods that use the transformer (by adding a version that doesn't have the transformer).